bin/p2-replicate/main.go

```diff
@@ -77,6 +87,27 @@ func main() {
 	replicator.Logger.Logger.Formatter = new(logrus.TextFormatter)
 	replicator.MinimumNodes = *minNodes
 
+	if *skip {
+		status, err := healthChecker.LookupHealth(manifest.ID())
+		if err != nil {
+			replicator.Logger.WithField("err", err).Errorln("Could not retrieve health checks to resolve skipped nodes")
+			os.Exit(1)
+		}
+
+		var newHosts []string
+		for _, host := range *hosts {
+			thisHealth, _ := status.ForNode(host)
+			if thisHealth.Health != health.Passing && thisHealth.Health != health.Unknown {
+				replicator.Logger.WithFields(logrus.Fields{
+					"host":   host,
+					"health": thisHealth.Health,
+				}).Infoln("Skipping unhealthy host")
+			}
+			newHosts = append(newHosts, host)
+		}
+		replicator.Allocation = allocation.NewAllocation(newHosts...)
```

Comment:
Skip ignores nodes that are strictly less than the threshold (ie the default value of the threshold is passing, so skip ignores warn/critical). I agree the semantics are rather confusing because of how the two flags interact... this was the most logical approach I could come up with.

I don't really see the point of having a skip at all if you would still update out-of-date unhealthy nodes. What's the intended use case for skipping unhealthy nodes? My impression was that we'd probably use this flag if a node was so damaged that it could not take a deploy, and the deployer doesn't want it to block a deployment to the other healthy nodes. Were there other ideas in mind?